### PR TITLE
Add spring-session-data-gemfire and spring-session-data-geode to managed BOM dependencies.

### PR DESCRIPTION
**Pull request by jxblum**
Resolves gh-1.

### CODE REVIEW

**Comment by jxblum** [comment]
Hi @vpavic -

Thank you for speedy follow-up and reply.  @rwinch and I discussed this internally in the #spring-session _Slack_ channel and @wilkinsona chimed in with...

> _As far as I can tell, Session Data GemFire and Session Data Geode only depend directly on modules that are already in the Platform so adding them is straightforward._

@rwinch can confirm.

Additionally, this wouldn't be any different than how _Spring Boot_ [pulls in (imports)](https://github.com/spring-projects/spring-boot/blob/v2.0.0.RELEASE/spring-boot-project/spring-boot-dependencies/pom.xml#L156) the _Spring Data Release Train_ Maven BOM file now, where the _SD Release Train Maven BOM_ file [includes](https://github.com/spring-projects/spring-data-build/blob/2.0.5.RELEASE/bom/pom.xml#L51-L70) both `spring-data-gemfire` and `spring-data-geode`.

In other words, the _Spring IO Platform_ [has everything it needs](https://github.com/spring-io/platform/blob/master/platform-bom/pom.xml#L706-L768) (and [this](https://github.com/spring-io/platform/blob/master/platform-bom/pom.xml#L498-L650)) to support both _Spring Data for Apache Geode_ and _Spring Data for Pivotal GemFire_.

As @wilkinsona mentions, neither `spring-session-data-geode` nor `spring-session-data-gemfire` pulls in any additional dependencies that the _Spring IO Platform_ doesn't already support.

Hopefully this make sense.  Let me know if there are additional questions; happy to help.

I will let @rwich chime in about the `spring-session-data-gemfire` POM behaving like a dependency aggregator.

Cheers.
